clean_build.sh

```diff
@@ -1,32 +1,74 @@
+    if command -v cmake 2>/dev/null; then
+        B_CMAKE=$(command -v cmake)
+        # We have a cmake executable available, now let's proceed!
+    else
+        # As self-explanatory, the cmake executable isn't available, so we should fail here.
+        echo "ERROR: CMake not in $PATH, cannot build! Please install CMake, or if this persists, file a bug report."
+        exit 1
+    fi
 fi
-# default build configuration
+
+# Set default build type, but in CI, this is set to Release.
+# For downstream distributions, our recommendation is to set this type in your build scripts to Release, and supply Debug build types in your debug packages.
+
 B_BUILD_TYPE=${B_BUILD_TYPE:-Debug}
+
 if [ "$(uname)" = "Darwin" ]; then
-    # OSX needs a lot of extra help, poor thing
-    # run the osx_environment.sh script to fix paths
-    . ./osx_environment.sh
-    B_CMAKE_FLAGS="-DCMAKE_OSX_SYSROOT=$(xcode-select --print-path)/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -DCMAKE_OSX_DEPLOYMENT_TARGET=10.9 $B_CMAKE_FLAGS"
+    # macOS needs a little help, so we source this environment script to fix paths.
+    if [ ! -e "./macOS_environment.sh" ]; then
+        echo "macOS environment script not found, this isn't meant to happen!"
+        exit 1
+    else
+        . ./macOS_environment.sh
+    fi
+
+    B_CMAKE_FLAGS="-DCMAKE_OSX_SYSROOT=$(xcode-select --print-path)/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -DCMAKE_OSX_DEPLOYMENT_TARGET=10.9 ${B_CMAKE_FLAGS:-}"
+fi
+
+# Source local build variables to the environment, if available.
+# If not, continue as normal, and silently.
+if [ -e "./build_env.sh" ]; then
+    . "./build_env.sh"
 fi
-# allow local customizations to build environment
-[ -r ./build_env.sh ] && . ./build_env.sh
 
 # Initialise Git submodules
 git submodule update --init --recursive
 
-B_CMAKE_FLAGS="-DCMAKE_BUILD_TYPE=$B_BUILD_TYPE $B_CMAKE_FLAGS"
-rm -rf build
-mkdir build || exit 1
-cd build || exit 1
-echo "Starting Barrier $B_BUILD_TYPE build..."
-$B_CMAKE $B_CMAKE_FLAGS .. || exit 1
-make || exit 1
-echo "Build completed successfully"
+B_CMAKE_FLAGS="-DCMAKE_BUILD_TYPE=${B_BUILD_TYPE} ${B_CMAKE_FLAGS:-}"
+
+# Clear build directory, but do a conditional first!
+
+if [ -d "./build" ]; then
+    rm -rf ./build
+fi
+
+# Previous versions of this script created the build directory, and CD'd into it - CMake allows us to do this another way...
+
+# Note: If you use Ninja (i.e, cmake -GNinja -B build), run Ninja like: "ninja -C build", just like you would with Meson.
+
+$B_CMAKE "$B_CMAKE_FLAGS" -B build || exit 1
+
+
+echo "INFO: Now commencing Barrier build process..."
+echo "INFO: We're building an $B_BUILD_TYPE output type."
+$(command -v make) -C build || exit 1
+
+# Implictly, we assume the build was successful due to no exits.
+# Later revisions of this script should do conditionals. TODO.
+
+echo "INFO: Success! The build completed successfully!"
+
+exit
```

Comment:
Redundant

Reply:
It depends on what your definition of 'Redudunant' is. I would say that it would make searching logs for the build being completed simpler, but I can see your point. I'll remove it for now, as we can revisit in the future if someone wants this output.

Reply:
Fixed in 250072e9.

Reply:
Still here somehow

Reply:
Ah, I was talking about the `exit`.

Reply:
You're quite right. Removed in 24f45e63.